ui/src/app/views/rename-modal/component.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+
+import {
+  Button
+} from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import {ReactComponent} from 'app/utils';
+
+
+interface RenameModalProps {
+  resource: any;
+  resourceType: string;
+  onRename: Function;
+  onCancel: Function;
+}
+interface RenameModalState {
+  loading: boolean;
+  newName: string;
+}
+
+@ReactComponent({
```

Comment:
@panentheos is there a symbol I can use with this approach to render the original RenameModal directly in React? Or would the idea for migration instead be to remove `@ReactComponent` and compose into other React components in one atomic change.